src/core/main.qml

```diff
@@ -47,121 +47,152 @@ Window {
     width: ApplicationSettings.previousWidth
     height: ApplicationSettings.previousHeight
     minimumWidth: 400 * ApplicationInfo.ratio
-    minimumHeight: 400 * ApplicationInfo.ratio
+    minimumHeight: 400 * ApplicationInfo.ratio 
     title: "GCompris"
-
+    
+    
     /// @cond INTERNAL_DOCS
-
+    
     property var applicationState: Qt.application.state
 
+    property var selectedMusicIndex 
+    property var rccBackgroundMusic: ApplicationInfo.getBackgroundMusicFromRcc();
+    property alias backgroundMusic: backgroundMusic
+    property bool isBackgroundMusicAllowed: ApplicationSettings.isBackgroundMusicEnabled
     onApplicationStateChanged: {
         if (ApplicationInfo.isMobile && applicationState != Qt.ApplicationActive) {
             audioVoices.stop();
             audioEffects.stop();
         }
     }
-
+    
     onClosing: Core.quit(main)
+   
+    onIsBackgroundMusicAllowedChanged: {
+        if(isBackgroundMusicAllowed) {
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/" + rccBackgroundMusic[selectedMusicIndex]));
+        }
+        else {
+            backgroundMusic.stop()
+        }
+    }
 
     GCAudio {
         id: audioVoices
         muted: !ApplicationSettings.isAudioVoicesEnabled
-
+        
         Timer {
```

Comment:
You added a lot of crap with such changes, i.e. inserted blanks at the beginning of the lines. Pls. don't, pls. revert these lines!
Hint: use qtcreator, which (configurably) removes trailing spaces/tabs upon save!